ops/terraform/services/server/modules/bfd_server_asg/main.tf

```diff
@@ -1,10 +1,75 @@
+    }
+    even = {
+      name             = "${aws_launch_template.main.name}-even"
+      desired_capacity = local.even_needs_scale_out ? max(local.odd_remote_desired_capacity, var.asg_config.desired) : (local.even_maintains_state ? local.even_remote_desired_capacity : 0)
+      max_size         = var.asg_config.max
+      min_size         = local.even_needs_scale_out ? max(local.odd_remote_desired_capacity, var.asg_config.desired) : (local.even_maintains_state ? local.even_remote_desired_capacity : 0)
+      warmpool_size    = local.even_needs_scale_out ? max(local.odd_remote_warmpool_min_size, var.asg_config.min) : (local.even_maintains_state ? local.even_remote_warmpool_min_size : 0)
+    }
+  }
+
   env      = terraform.workspace
   seed_env = var.seed_env
 
   # When the CustomEndpoint is empty, fall back to the ReaderEndpoint
   rds_reader_endpoint = data.external.rds.result["ReaderEndpoint"]
 
+  latest_ltv                     = tonumber(aws_launch_template.main.latest_version)
+  is_latest_launch_template_odd  = local.latest_ltv % 2 == 1
+  is_latest_launch_template_even = local.latest_ltv % 2 == 0
+
+  odd_remote_desired_capacity  = tonumber(data.external.current_asg.result["odd_desired_capacity"])
+  even_remote_desired_capacity = tonumber(data.external.current_asg.result["even_desired_capacity"])
+
+  odd_remote_warmpool_min_size  = tonumber(data.external.current_asg.result["odd_warmpool_min_size"])
+  even_remote_warmpool_min_size = tonumber(data.external.current_asg.result["even_warmpool_min_size"])
+
+  #ODD scales OUT when launchtemplate is ODD and ODD ASG has 0 desired capacity
+  odd_needs_scale_out = alltrue([
+    local.is_latest_launch_template_odd,
+    local.odd_remote_desired_capacity == 0
+  ])
+
+  #ODD scales IN when launchtemplate EVEN, both ODD and EVEN ASGS have capacity
+  odd_needs_scale_in = alltrue([
+    local.is_latest_launch_template_even,
+    local.odd_remote_desired_capacity > 0,
+    local.even_remote_desired_capacity > 0
+  ])
+
+  #ODD maintains state when ODD does not need to scale IN or OUT
+  odd_maintains_state = alltrue([
+    !local.odd_needs_scale_out,
+    !local.odd_needs_scale_in
+  ])
+
+  #EVEN scales OUT when launchtemplate is EVEN and EVEN ASG has 0 desired capacity
+  even_needs_scale_out = alltrue([
+    local.is_latest_launch_template_even,
+    local.even_remote_desired_capacity == 0
+  ])
+
+  #EVEN scales IN when launchtemplate is ODD, both EVEN and ODD ASGs have capacity
+  even_needs_scale_in = alltrue([
+    local.is_latest_launch_template_odd,
+    local.even_remote_desired_capacity > 0,
+    local.odd_remote_desired_capacity > 0
+  ])
+
+  #EVEN maintains state when EVEN does not need to scale IN or OUT
+  even_maintains_state = alltrue([
+    !local.even_needs_scale_out,
+    !local.even_needs_scale_in
+  ])
+
```

Comment:
```suggestion
  # We need to know when an operator is attempting to create a new Launch Template version (most
  # likely with a new AMI) when both odd and even are scaled-out. This is an invalid state, and
  # likely indicates that the operator is attempting to resolve the situation where "green"
  # instances failed the Regression Suite or other verification before being promoted to "blue"
  # instances. Because "green" and "blue" changes based on Launch Template version divisiblity, the
  # operator must resolve this situation out-of-band (likely by incrementing the LT version and then
  # resetting the previous green ASG) before applying.
  prev_green_needs_reset = alltrue([
    # ODD is scaled-out
    local.odd_remote_desired_capacity > 0,
    # EVEN is scaled-out
    local.even_remote_desired_capacity > 0,
    # But, the existing, latest launch template is being replaced by a new Template (likely with a
    # new AMI) indicating that an operator is attempting to create new instances in-between green
    # (the incoming version) being accepted into blue
    data.aws_launch_template.main.latest_version != aws_launch_template.main.latest_version
  ])

```